webviz_subsurface/plugins/_relative_permeability.py

```diff
@@ -700,37 +821,157 @@ def add_realization_traces(df, color_by, curves, sataxis, colors, nplots):
     return traces
 
 
-def add_scal_traces(df, curves, sataxis, nplots):
+# pylint: disable=too-many-locals
+def add_scal_traces(
+    df, color_by, curves, sataxis, colors, nplots, scenario_name, plugin_mode
+):
     """Renders scal recommendation traces"""
     traces = []
+    linestyle = {"pess": "dash", "base": "solid", "opt": "dot"}
     for curve_no, curve in enumerate(
         [curve for curve in curves if curve in df.columns]
     ):
         yaxis = "y" if nplots == 1 or curve.startswith("KR") else "y2"
         xaxis = "x" if yaxis == "y" else "x2"
-        traces.extend(
-            [
-                {
-                    "type": "scatter",
-                    "x": case_df[sataxis],
-                    "y": case_df[curve],
-                    "xaxis": xaxis,
-                    "yaxis": yaxis,
-                    "hovertext": (
-                        f"{curve}, Satnum: {satnum}<br>" f"{case.lower().capitalize()}"
-                    ),
-                    "name": "SCAL",
-                    "legendgroup": "SCAL",
-                    "marker": {
-                        "color": "black",
-                    },
-                    "line": {"dash": "dash"},
-                    "showlegend": curve_no == 0 and satnum_no == 0 and case_no == 0,
-                }
-                for satnum_no, (satnum, satnum_df) in enumerate(df.groupby("SATNUM"))
-                for case_no, (case, case_df) in enumerate(satnum_df.groupby("CASE"))
-            ]
-        )
+        if plugin_mode == "scal_scenarios":
+            # traces.extend(
+            #     [
+            #         {
+            #             "type": "scatter",
+            #             "x": case_df[sataxis],
+            #             "y": case_df[curve],
+            #             "xaxis": xaxis,
+            #             "yaxis": yaxis,
+            #             "hovertext": (
+            #                 f"{curve}, Satnum: {satnum}<br>" f"{case.lower().capitalize()}"
+            #             ),
+            #             "name": scenario_name,
+            #             "legendgroup": scenario_name,
+            #             "marker": {
+            #                 "color": "black",
+            #             },
+            #             "line": {"dash": linestyle.get(case, "solid")},
+            #             "showlegend": curve_no == 0 and satnum_no == 0 and case_no == 0,
+            #         }
+            #         for satnum_no, (satnum, satnum_df) in enumerate(df.groupby("SATNUM"))
+            #         for case_no, (case, case_df) in enumerate(satnum_df.groupby("CASE"))
+            #     ]
+            # )
+            ##
```

Comment:
Add a comment in the comment on when this (potentially) will be needed again? If not - remove?

Reply:
haha, yeah that should have been removed...